packages/opencensus-core/src/trace/model/span-base.ts

```diff
@@ -183,6 +183,18 @@ export abstract class SpanBase implements types.Span {
     } as types.MessageEvent);
   }
 
+  /**
+   * Sets a status to the span.
+   * @param code The status code.
+   * @param message optional A developer-facing error message.
+   */
+  setStatus(code: number, message?: string) {
+    this.status.code = code;
+    if (message) {
```

Comment:
If someone sets a new code but no message, would this check leave the old message assigned to status?

Could this function just be `this.status = {code, message}`, since we the `message` field is optional?

Reply:
Good catch! Done.